### DISCUSSION TITLE
RabbitMQ Auth0 throws error {error,key_not_found}

### DISCUSSION THREAD

**Discussion by sandeep6011382** [rabbitmq/rabbitmq-server#11346] [Questions] [answered]
Hi,

I am trying to integrate RabbitMQ with Auth0. After going through all the documents added the required configurations and when testing it is throwing below error.

2024-05-23 23:09:09.812203-07:00 [debug] <0.5370.0> Authentication using an OAuth 2/JWT token failed: {error,key_not_found}
2024-05-23 23:09:09.812367-07:00 [debug] <0.5370.0> User 'kyBZyeD5jDcx1RHBrRb3cI0ubzx24u8B' failed authentication by backend rabbit_auth_backend_oauth2
2024-05-23 23:09:09.812450-07:00 [debug] <0.5370.0> User 'kyBZyeD5jDcx1RHBrRb3cI0ubzx24u8B' failed authentication by backend rabbit_auth_backend_internal
2024-05-23 23:09:09.812542-07:00 [warning] <0.5370.0> HTTP access denied: user 'kyBZyeD5jDcx1RHBrRb3cI0ubzx24u8B' - invalid credentials

RabbitMQ version :: 3.12.13
Erlang version :: 26.2.4

RabbitMQ config.

auth_backends.1 = rabbit_auth_backend_oauth2
auth_backends.2 = rabbit_auth_backend_internal
auth_oauth2.https.peer_verification = verify_none
auth_oauth2.resource_server_id=rabbitmq
auth_oauth2.additional_scopes_key=permissions
auth_oauth2.https.cacertfile=/apps/rabbitmq/etc/rabbitmq/rabbitmq-non-prod.pem
management.oauth_enabled=true
management.oauth_client_id=<<clientId>>
management.oauth_provider_url=https://<<tenant>>
management.oauth_scopes = openid profile rabbitmq.tag:administrator
auth_oauth2.signing_keys.id1 = /apps/rabbitmq/etc/rabbitmq/public.pem
auth_oauth2.algorithms.1 = RS256
auth_oauth2.default_key = id1

JWT token ::
eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6Ijd2eUJqNGllbnpxV043OWVGM2dqbiJ9.eyJpc3MiOiJodHRwczovL2Rldi1jcndrNnIzeWJrODc1emgxLnVzLmF1dGgwLmNvbS8iLCJzdWIiOiJnb29nbGUtb2F1dGgyfDEwNTgxNDkzMzg0NDc5NTk1MzgyNiIsImF1ZCI6WyJyYWJiaXRtcSIsImh0dHBzOi8vZGV2LWNyd2s2cjN5Yms4NzV6aDEudXMuYXV0aDAuY29tL3VzZXJpbmZvIl0sImlhdCI6MTcxNjUzMDk0OSwiZXhwIjoxNzE2NjE3MzQ5LCJzY29wZSI6Im9wZW5pZCBwcm9maWxlIHJhYmJpdG1xLnRhZzphZG1pbmlzdHJhdG9yIiwiYXpwIjoia3lCWnllRDVqRGN4MVJIQnJSYjNjSTB1Ynp4MjR1OEIiLCJwZXJtaXNzaW9ucyI6WyJyYWJiaXRtcS5jb25maWd1cmU6Ki8qIiwicmFiYml0bXEucmVhZDoqLyoiLCJyYWJiaXRtcS50YWc6YWRtaW5pc3RyYXRvciIsInJhYmJpdG1xLndyaXRlOiovKiJdfQ.IHbLrZL-CzYInguzGWY7Qs4OcXcb0HYwi7qAxoRSi1nWmy0Rc5RD8iCW1jUlhIhPYtASnQ4seyoCMWHNy6EIbj1GG2j0t8LhoR0j-AblIKi6wsCzfFtAfo_25xzsr61gcqGDr49MFSRl7H_ZTFb3ol_0bMbD8kJh5gfS7AThuREFFlC9p4_jCJBzYi4sDVmXi2XQbDBF0WmbcvfC-6LdXPCYcbPgB9If_M5nWOyJa3Kul53Bzgjxc_Al2ttMa4JcAPuyUyb-G6f7UmaUrDyKruUjn6REH_xxWsAND5dg1wMb5g4tmDfqQUu7awYQet5Vjccy94fiWIQslebyKRQfJQ

Please help me to resolve the issue.

Thanks & Regards,
Sandeep

**Comment by lukebakken** [comment, marked as answer]
Hello,

Thank you for using RabbitMQ. We have recently implemented the following community support policy, which excludes general, free support for OAuth2 questions:

https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md#exceptions-question-that-will-be-ignored

> Guidance and investigations related to these features will only be available to customers with VMware Tanzu RabbitMQ commercial licenses.

**Reply by michaelklishin** [reply]
Plus the series used is 3.12.x, which is not the latest.
